lib/backend/repository/repository2.go

```diff
@@ -0,0 +1,254 @@
+package repository
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"strconv"
+	"time"
+
+	"github.com/appc/docker2aci/lib/common"
+	"github.com/appc/docker2aci/lib/types"
+	"github.com/appc/docker2aci/lib/util"
+	"github.com/appc/spec/schema"
+	"github.com/coreos/ioprogress"
+)
+
+const (
+	defaultV2IndexURL = "registry-1.docker.io"
+)
+
+type v2Manifest struct {
+	Name     string `json:"name"`
+	Tag      string `json:"tag"`
+	FSLayers []struct {
+		BlobSum string `json:"blobSum"`
+	} `json:"fsLayers"`
+	History []struct {
+		V1Compatibility string `json:"v1Compatibility"`
+	} `json:"history"`
+	//Signature []byte `json:"signature"`
+}
+
+func (rb *RepositoryBackend) supportsV2(indexURL string) (bool, error) {
```

Comment:
The [API spec](https://docs.docker.com/registry/spec/api/) mentions this:

> When a 200 OK or 401 Unauthorized response is returned, the “Docker-Distribution-API-Version” header should be set to “registry/2.0”. Clients may require this header value to determine if the endpoint serves this API. When this header is omitted, clients may fallback to an older API version.

I understand that we're not supporting auth for now but maybe we should check that header although a 200 OK is probably a good enough indication that v2 is supported.